ingester/src/data.rs

```diff
@@ -956,6 +1069,22 @@ impl DataBuffer {
         Ok(())
     }
 
+    /// Return a QueryableBatch of the persisting batch after applying new tombstones
+    pub fn get_persisting_data(&self) -> Result<Option<QueryableBatch>> {
```

Comment:
I don't see anywhere that this function would return an `Err` type; does it still need to return a `Result` or could this be:

```suggestion
    pub fn get_persisting_data(&self) -> Option<QueryableBatch> {
```

?

Reply:
Yes, the result is using to return the right error message to the cline (which is Querier). This function invokes the one you commented above https://github.com/influxdata/influxdb_iox/pull/3825#discussion_r816044742